packages/webdriverio/src/commands/browser/react$$.ts

```diff
@@ -29,13 +29,18 @@ import fs from 'fs'
 import { enhanceElementsArray } from '../../utils'
 import { getElements } from '../../utils/getElementObject'
 import { waitToLoadReact, react$$ as react$$Script } from '../../scripts/resq'
+import type { ElementReference } from '../../types'
 
 const resqScript = fs.readFileSync(require.resolve('resq'))
 
-export default async function react$$ (selector, { props = {}, state = {} } = {}) {
+export default async function react$$ (
+    this: WebdriverIO.BrowserObject,
+    selector: string,
+    { props = {}, state = {} }: WebdriverIO.ReactSelectorOptions = {}
+) {
     await this.executeScript(resqScript.toString(), [])
     await this.execute(waitToLoadReact)
-    const res = await this.execute(react$$Script, selector, props, state)
+    const res = await this.execute(react$$Script, selector, props, state) as ElementReference[]
```

Comment:
if `execute` returns either `ElementReference` or `ElementReference[]`, I'd suggest adding `ElementReference | ElementReference[]` as the return type for `execute` and then using a type check here